pandas/tests/groupby/aggregate/test_aggregate.py

```diff
@@ -115,6 +117,69 @@ def test_groupby_aggregation_mixed_dtype():
     tm.assert_frame_equal(result, expected)
 
 
+def test_groupby_aggregation_non_numeric_dtype():
```

Comment:
Move all tests to test_function, there is a class there TestNumericOnly. test_aggregate is meant for tests on e.g. DataFrameGroupBy.aggregate.